examples/config-policy.json

```diff
@@ -14,6 +14,7 @@
       "failDelay": 1
     },
     "accessControl": {
+      "anonymous": false,
       "**": {

```

Comment:
This "**" is the default right? Should "anonymous" move here instead?
And can "anonynous" be set per repo-policy?

Reply:
Hei Ram, from what I see in the code we can do this in authz

basically we bypass authn if accessControl != nil and in authz decide if we throw authFail() based on anonymousAccess  flag combined with the condition that username is nil.

The problem arise in certificate authentication, this flag tls.RequireAndVerifyClientCert is used when we start the server.
What to do in this case? currently we look at the global flag and decide based on it, but per repo flags are used in the context of a request, not the whole server.

If we look at "**"'s anonymousAccess it will not be consistent, we can not authFail for a certain repo which has the anonymousAccess on false in case of certificates auth.

hope this is clear enough :D.

Reply:
We spoke about this and we want to use one global bool flag in http.auth config, basically the same as allowReadAccess and based on it to decide if we let anonymous access with certs authentication. related issue: #614 

And use a per repo anonymous policy for request scoped authorization.

example:
```
      "**": {
        "anonymousAccess": ["read"],
        "policies": [
          {
            "users": [
              "charlie", "bob"
            ],
            "actions": [
              "read",
              "create",
              "update"
            ]
          }
        ],
        "defaultPolicy": [
          "read",
          "create"
        ]
      },
```

authenticated users charlie and bob can do read, cread, update (users policy)
authenticated users others than charlie and bob can do read, create (default policy)
unauthenticated users can do read (anonymous policy)

Reply:
Why not write a:

```
func allowReadAccess()  {
// iterate over all policies, and if there is any one with anonymous read access, then return true here
// and when we know the URL and method, we can enforce specifically
}

Reply:
this will not work for TLS mutual auth.

either we make it optional or required at server startup, from what I understand, besides this, it would work...

Reply:
ok, you are right :D, we make tls certs optional, but we verify them at request level, got it

Reply:
Let's resolve this